hub/core/tensor.py

```diff
@@ -117,6 +130,49 @@ def add_samples_to_tensor(
     write_index_map(key, storage, index_map)
 
 
+def add_index_map_to_tensor(
+    index_map_dict: dict,
+    key: str,
+    storage: StorageProvider,
+):
+    """Adds samples to a tensor that already exists. bytes of index_map are chunked and sent to `storage`.
+    For more on chunking, see the `generate_chunks` method.
+
+    Args:
+        index_map_dict (dict): Bytes of the image that should be chunked and written to the storage.
+        key (str): Key for where the chunks, index_map, and meta will be located in `storage` relative to it's root.
+        storage (StorageProvider): StorageProvider for storing the chunks, index_map, and meta.
+
+    Raises:
+        TensorDoesNotExistError: If a tensor at `key` does not exist. A tensor must be created first using
+            `create_tensor(...)`.
+    """
+    if not tensor_exists(key, storage):
+        raise TensorDoesNotExistError(key)
+
+    index_map = read_index_map(key, storage)
+    tensor_meta = read_tensor_meta(key, storage)
+    shape = index_map_dict["shape"]
+    if "min_shape" not in tensor_meta:
+        tensor_meta["dtype"] = str(index_map_dict["dtype"])
+        tensor_meta["min_shape"] = shape
+        tensor_meta["max_shape"] = shape
+    index_map_entry = {"chunk_names": []}  # type: ignore
+    index_map_entry = write_bytes(
+        index_map_dict["bytes"], key, tensor_meta["chunk_size"], storage, index_map
+    )
+    index_map_entry["compression"] = index_map_dict["compression"]
```

Comment:
nit-pick: would love all these keys like `compression` `dtype`,`shape` etc. to be stored as  stringconstants somewhere.